ambry-clustermap/src/test/java/com.github.ambry.clustermap/ClusterMapUtilsTest.java

```diff
@@ -252,16 +286,26 @@ private void verifyWritablePartitionsReturned(ClusterMapUtils.PartitionSelection
     ((MockReplicaId) testedPart1.getReplicaIds().get(0)).setSealedState(true);
     allPartitionIds.remove(testedPart1);
     expectedReturnForClassBeingTested.remove(testedPart1);
-    assertCollectionEquals("Partitions returned not as expected", allPartitionIds, psh.getWritablePartitions(null));
+    assertCollectionEquals("Partitions returned not as expected", allPartitionIds,
+        psh.getWritablePartitions(null));
     assertCollectionEquals("Partitions returned not as expected", expectedReturnForClassBeingTested,
         psh.getWritablePartitions(classBeingTested));
+    assertInCollection("Partitions returned not as expected", allPartitionIds,
+        psh.getRandomWritablePartition(null, null));
+    assertInCollection("Partitions returned not as expected", expectedReturnForClassBeingTested,
+        psh.getRandomWritablePartition(classBeingTested, null));
     // all READ_ONLY
     ((MockReplicaId) testedPart2.getReplicaIds().get(0)).setSealedState(true);
     allPartitionIds.remove(testedPart2);
     expectedReturnForClassBeingTested.remove(testedPart2);
-    assertCollectionEquals("Partitions returned not as expected", allPartitionIds, psh.getWritablePartitions(null));
+    assertCollectionEquals("Partitions returned not as expected", allPartitionIds,
+        psh.getWritablePartitions(null));
     assertCollectionEquals("Partitions returned not as expected", expectedReturnForClassBeingTested,
         psh.getWritablePartitions(classBeingTested));
+    assertInCollection("Partitions returned not as expected", allPartitionIds,
+        psh.getRandomWritablePartition(null, null));
+    assertNull("Partitions returned not as expected",
+        psh.getRandomWritablePartition(classBeingTested, null));
```

Comment:
This method contains some duplicated assertions. For example: line241~248, line255~262, line268~275 and line289~296. How about moving them to a separate method and simplify this test?

Reply:
sure.

Reply:
fixed.